### ISSUE TITLE
Severe browser slowdown when increasing number of iterations

### ISSUE THREAD

**Issue by LiakGamer** [seriallos/raidbots-issues#334]
A very severe slowdown happens when increasing the number of iterations on top gear, the slowdown is gradual as more items get selected. This seems tp be especially pronounced on rings and trinkets as these create more possible combinations. I got to 80 million iterations before I tried to disable some gem options and it froze my browser.

Running on Chrome version 101.0.4951.41 (Official Build) (64-bit)

Also happens on Firefox. Can't check current version as I don't have it installed.

Running on 16gbs of ram. 9gbs free at the moment of testing. Not many tabs were opened.

CPU is an i5 11400f. (Don't know if it matters just giving you as much as I can).

I just cleared my hard drive. I had the issue before and it persisted. Drivers are up to date as far as I'm aware.

**Comment by seriallos** [comment]
Yeah, there are known significant performance problems when using "Replace Existing Gems/Enchants" while selecting a large number of items. Realistically, it's unlikely that these will be fixed in the short/medium term due to some deep  and complex technical issues that would need to be handled.

If your browser is completely unable to load Top Gear, you can use the reset tool which should get things working again: https://www.raidbots.com/reset

Otherwise, you just need to reduce the number of selections before the performance problem kicks in. This is usually fine as long as you stay within ~16 million iterations which is the maximum Raidbots actually allows you to run.
